namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/listeners/AssociationListener.java

```diff
@@ -50,7 +50,10 @@ public void OnPlayerJoin(PlayerJoinEvent event) {
 	public void OnPlayerLogin(PlayerLoginEvent event) {
 		final String name = associations.getCurrentName(event.getPlayer().getUniqueId());
 		if (name == null)
+		{
+			associations.addPlayer(event.getPlayer().getName(), event.getPlayer().getUniqueId());
 			return;
```

Comment:
hmm so this is almost there

if name is null, need to add to associations, then replace name with name = getcurrent name; get rid of the return, so that setplayerprofile replaces the profile's version of name with the new name from getcurrent name, etc.